### PR TITLE
build(cli): allow passing in-stream using command.Cli

### PR DESCRIPTION
**Pull request by distorhead**
Use command.Cli::In() if it is set, use os.Stdin otherwise.

### CODE REVIEW

**Comment by jedevc** [comment]
Sorry, I don't have any context for this change :eyes: Could you explain the change a bit more, and what it resolves?

**Comment by distorhead** [comment]
Rebased on actual master branch to resolve conflict.

File: commands/build.go [hunk 1]
```diff
@@ -832,3 +832,10 @@ func printValue(printer printFunc, version string, format string, res map[string
 	}
 	return printer([]byte(res["result.json"]), os.Stdout)
 }
+
+func getInStream(dockerCli command.Cli) io.ReadCloser {
+	if dockerCli.In() != nil {
```
Comment (crazy-max) [inline commands/build.go:837]: Don't think this is needed, `In()` is always set in cli: 

* https://github.com/docker/buildx/blob/341fb65f6f568696ed5e2585eb747aedaed7ee1a/cmd/buildx/main.go#L54
* https://github.com/docker/cli/blob/5be21394cbbafc5627bb3e8a54014c108a534b52/cli/command/cli.go#L494
* https://github.com/docker/cli/blob/master/cli/command/cli_options.go#L16C1-L26

File: commands/build.go [hunk 2]
```diff
@@ -324,7 +324,7 @@ func runControllerBuild(ctx context.Context, dockerCli command.Cli, opts *contro
 	var retErr error
 	var resp *client.SolveResponse
 	f := ioset.NewSingleForwarder()
-	f.SetReader(os.Stdin)
+	f.SetReader(getInStream(dockerCli))
```
Comment (crazy-max) [inline commands/build.go:327]:
```suggestion
	f.SetReader(dockerCli.In())
```

File: commands/build.go (hunk 1 above)
Comment (distorhead) [reply commands/build.go]: Thanks for a catch! Actualized code.

**Comment by distorhead** [comment]
@crazy-max Hi,
Done here, just resolved conflict and rebased on actual upstream branch.

**Review by jedevc — APPROVED** [review]

**Review by crazy-max — APPROVED** [review]

**Comment by distorhead** [comment]
> @distorhead I've pushed some additional commits to apply this same logic across the whole `commands` package. Both `bake` and `debug-shell` should also do the same.

Thanks a lot!

**Pull request merged**